### ISSUE TITLE
Wired error on combining stage

### ISSUE THREAD

**Issue by realsarm** [tilmanmoser/bbb-video-download#25]
./node12/bin/node index.js -i /var/bigbluebutton/published/presentation/62a824f136d6ad592ed60e90d2a66e2b61c13438-1603955022
585 -o video.mp4                                                                
[Start] Rendering downloadable video for BBB presentation  {                    
  args: Namespace(input='/var/bigbluebutton/published/presentation/62a824f136d6a
d592ed60e90d2a66e2b61c13438-1603955022585', output='video.mp4'),                
  env: { chunksize: 25 }                                                        
}                                                                               
[AVFilterGraph @ 0x55e75b4b7a80] No output pad can be associated to link label '
d2'.                                                                            
Error initializing complex filters.                                             
Invalid argument                                                                
[Error] Failed rendering downloable video for BBB presentation  {               
  args: Namespace(input='/var/bigbluebutton/published/presentation/62a824f136d6a
d592ed60e90d2a66e2b61c13438-1603955022585', output='video.mp4'),                
  env: { chunksize: 25 },                                                       
  workdir: './tmp/datajy2XHq'                                                   
} Error: Command failed: ffmpeg -hide_banner -loglevel error -i ./tmp/datajy2XHq
/slides.mp4 -i /var/bigbluebutton/published/presentation/62a824f136d6ad592ed60e9
0d2a66e2b61c13438-1603955022585/deskshare/deskshare.webm -filter_complex_script 
./tmp/datajy2XHq/filters.txt -map '[v6]' -threads 1 -strict -2 ./tmp/datajy2XHq/
presentation.tmp.mp4                                                            
[AVFilterGraph @ 0x55e75b4b7a80] No output pad can be associated to link label '
d2'.                                                                            
Error initializing complex filters.                                             
Invalid argument                                                                
                                                                                
    at checkExecSyncError (child_process.js:630:11)                             
    at Object.execSync (child_process.js:666:15)                                
    at combinedSlidesAndDeskshares (/opt/bbb-video-download/modules/presentation
.js:53:18)                                                                      
    at module.exports.createPresentationVideo (/opt/bbb-video-download/modules/p
resentation.js:22:18)                                                           
    at async module.exports.createVideo (/opt/bbb-video-download/modules/process
or.js:20:26)                                                                    
    at async run (/opt/bbb-video-download/index.js:10:9) {                      
  status: 1,                                                                    
  signal: null,                                                                 
  output: [                                                                     
    null,                                                                       
    <Buffer >,                                                                  
    <Buffer 5b 41 56 46 69 6c 74 65 72 47 72 61 70 68 20 40 20 30 78 35 35 65 37
 35 62 34 62 37 61 38 30 5d 20 4e 6f 20 6f 75 74 70 75 74 20 70 61 64 20 63 61 6
e ... 88 more bytes>                                                            
  ],                                                                            
  pid: 16451,                                                                   
  stdout: <Buffer >,                                                            
  stderr: <Buffer 5b 41 56 46 69 6c 74 65 72 47 72 61 70 68 20 40 20 30 78 35 35
 65 37 35 62 34 62 37 61 38 30 5d 20 4e 6f 20 6f 75 74 70 75 74 20 70 61 64 20 6
3 61 6e ... 88 more bytes>

I'm getting this error.
How can I resolve it?

**Comment by realsarm** [comment]
This is its filter.txt
`[1]scale=800:-2,split[d0][d1][d2][d3][d4];
[0][d0]overlay=enable='between(t,74.1,432.2)'[v2];
[v2][d1]overlay=enable='between(t,466.7,555.8)'[v3];
[v3][d2]overlay=enable='between(t,647.5,796.2)'[v4];
[v4][d3]overlay=enable='between(t,1829.5,1867.1)'[v5];
[v5][d4]overlay=enable='between(t,3020.8,4115)'[v6]`

**Comment by tilmanmoser** [comment]
Please retest with v1.1.3

**Comment by realsarm** [comment]
I'm now getting this
```

tart] Rendering downloadable video for BBB presentation  {                                             
  args: Namespace(input='/var/bigbluebutton/published/presentation/62a824f136d6ad592ed60e90d2a66e2b61c134
38-1603955022585', output='video.mp4')                                                                   
}                                                                                                        
Killed                                                                                                   
[Error] Failed rendering downloable video for BBB presentation  {                                        
  args: Namespace(input='/var/bigbluebutton/published/presentation/62a824f136d6ad592ed60e90d2a66e2b61c134
38-1603955022585', output='video.mp4'),                                                                  
  workdir: './tmp/databTeU6c'                                                                            
} Error: Command failed: ffmpeg -hide_banner -loglevel error -i ./tmp/databTeU6c/slides.mp4 -i /var/bigbl
uebutton/published/presentation/62a824f136d6ad592ed60e90d2a66e2b61c13438-1603955022585/deskshare/deskshar
e.webm -filter_complex_script ./tmp/databTeU6c/filters.txt -map '[v5]' -threads 1 -strict -2 ./tmp/databT
eU6c/presentation.tmp.mp4                                                                                
Killed                                                                                                   
                                                                                                         
    at checkExecSyncError (child_process.js:630:11)                                                      
    at Object.execSync (child_process.js:666:15)                                                         
    at combinedSlidesAndDeskshares (/opt/bbb-video-download/modules/presentation.js:54:18)               
    at module.exports.createPresentationVideo (/opt/bbb-video-download/modules/presentation.js:22:18)    
    at async module.exports.createVideo (/opt/bbb-video-download/modules/processor.js:20:26)             
    at async run (/opt/bbb-video-download/index.js:10:9) {                                               
  status: 137,                                                                                           
  signal: null,                                                                                          
  output: [ null, <Buffer >, <Buffer 4b 69 6c 6c 65 64 0a> ],                                            
  pid: 4570,                                                                                             
  stdout: <Buffer >,                                                                                     
  stderr: <Buffer 4b 69 6c 6c 65 64 0a>


```

**Comment by realsarm** [comment]
```

[1]trim=74.1:432.2[d0];[d0][0]scale2ref=oh*mdar:ih[do0][s0];[s0][do0]overlay=enable='between(t,74.1,432.2
)'[v1];                                                                                                  
[1]trim=466.7:555.8[d1];[d1][v1]scale2ref=oh*mdar:ih[do1][s1];[s1][do1]overlay=enable='between(t,466.7,55
5.8)'[v2];                                                                                               
[1]trim=647.5:796.2[d2];[d2][v2]scale2ref=oh*mdar:ih[do2][s2];[s2][do2]overlay=enable='between(t,647.5,79
6.2)'[v3];                                                                                               
[1]trim=1829.5:1867.1[d3];[d3][v3]scale2ref=oh*mdar:ih[do3][s3];[s3][do3]overlay=enable='between(t,1829.5
,1867.1)'[v4];                                                                                           
[1]trim=3020.8:4115[d4];[d4][v4]scale2ref=oh*mdar:ih[do4][s4];[s4][do4]overlay=enable='between(t,3020.8,4
115)'[v5]s


```

**Comment by realsarm** [comment]
This is the dmesg
```


[390559.550989] Out of memory: Kill process 5715 (ffmpeg) score 240 or sacrifice child
[390559.551244] Killed process 5715 (ffmpeg) total-vm:8281060kB, anon-rss:6174328kB, file-rss:0kB


```

**Comment by tilmanmoser** [comment]
Memory issue was fixed in v1.2.1.

**Comment by LiohMoeller** [comment]
I am running 1.2.1 and still got the memory issue:

[16156185.096621] Out of memory: Killed process 12416 (ffmpeg) total-vm:13932972kB, anon-rss:13846924kB, file-rss:0kB, shmem-rss:0kB, UID:997 pgtables:27300kB oom_score_adj:0

```
Error: Command failed: ffmpeg -hide_banner -loglevel error -f concat -i ./tmp/dataofGLjb/slides.txt -threads 1 -y -filter_complex "[0:v]fps=24, scale=800:-2[out]" -map '[out]' -strict -2 -crf 22 -pix_fmt yuv420p ./tmp/dataofGLjb/slides.mp4
    at checkExecSyncError (child_process.js:635:11)
    at Object.execSync (child_process.js:671:15)
    at renderVideo (/home/bigbluebutton/modules/slides.js:317:18)
    at module.exports.renderSlides (/home/bigbluebutton/modules/slides.js:13:15)
    at async module.exports.createPresentationVideo (/home/bigbluebutton/modules/presentation.js:10:20)
    at async module.exports.createVideo (/home/bigbluebutton/modules/processor.js:20:26)
    at async run (/home/bigbluebutton/index.js:10:9) {
  status: null,
  signal: 'SIGKILL',
  output: [ null, <Buffer >, <Buffer > ],
  pid: 113,
  stdout: <Buffer >,
  stderr: <Buffer >
```
sudo -u bigbluebutton docker-compose run --rm --user 997:995 app node index.js -v

1.2.1
